crates/polars-io/src/parquet/read/read_impl.rs

```diff
@@ -830,7 +833,7 @@ fn rg_to_dfs_par_over_rg(
                 if let Some(rc) = &row_index {
                     df.with_row_index_mut(
                         rc.name.clone(),
-                        Some(row_count_start as IdxSize + rc.offset),
+                        Some(row_count_start as IdxSize + rc.offset + slice.0 as IdxSize),
```

Comment:
In `fn rg_to_dfs*`, we also add `slice.0` (offset), as it may be non-zero in a negative-slice case